addons/payment/models/payment_transaction.py

```diff
@@ -651,47 +651,53 @@ def _set_authorized(self, state_message=None):
         :return: The updated transactions.
         :rtype: recordset of `payment.transaction`
         """
-        allowed_states = ('draft', 'pending')
+        allowed_states = ['draft', 'pending']
         target_state = 'authorized'
         txs_to_process = self._update_state(allowed_states, target_state, state_message)
         txs_to_process._log_received_message()
         return txs_to_process
 
-    def _set_done(self, state_message=None):
+    def _set_done(self, state_message=None, extra_allowed_state=None):
         """ Update the transactions' state to `done`.
 
         :param str state_message: The reason for setting the transactions in the state `done`.
         :return: The updated transactions.
         :rtype: recordset of `payment.transaction`
         """
-        allowed_states = ('draft', 'pending', 'authorized', 'error', 'cancel')  # 'cancel' for Payulatam
-        target_state = 'done'
+        allowed_states = ['draft', 'pending', 'authorized', 'error']
+        if extra_allowed_state:
+            allowed_states.append(extra_allowed_state)
+        target_state = 'done'  # 'cancel' for Payulatam
```

Comment:
This comment has nothing to do here =] It'll be managed in `payment_payulatam` from now on